source/common/stats/tag_extractor_impl.cc

```diff
@@ -49,7 +48,7 @@ std::string TagExtractorImpl::extractRegexPrefix(absl::string_view regex) {
 
 TagExtractorPtr TagExtractorImpl::createTagExtractor(const std::string& name,
                                                      const std::string& regex,
-                                                     const std::string& substr) {
+                                                     const std::string& substr, bool use_re2) {
```

Comment:
Can we use an enum instead of a bool to make the behavior clearer at the call site?